packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts

```diff
@@ -1,70 +1,201 @@
 
-/**
- * @deprecated Use `comboButtonClassNames.root` instead.
- */
-export const comboButtonClassName = 'fui-ComboButton';
 export const comboButtonClassNames: SlotClassNames<ComboButtonSlots> = {
   root: 'fui-ComboButton',
   content: 'fui-ComboButton__content',
   expandIcon: 'fui-ComboButton__expandIcon',
 };
 
+/*
+ * TODO: a number of spacing and animation values are shared with other form controls.
+ * We should probably find a way to share these values between form controls in the theme.
+ */
+
+const horizontalSpacing = {
+  xxs: '2px',
+  xs: '4px',
+  sNudge: '6px',
+  s: '8px',
+  mNudge: '10px',
+  m: '12px',
+};
+
+const iconSizes = {
+  small: '16px',
+  medium: '20px',
+  large: '24px',
+};
+
+const motionDurations = {
+  ultraFast: '0.05s',
+  normal: '0.2s',
+};
+
+const motionCurves = {
+  accelerateMid: 'cubic-bezier(0.7,0,1,0.5)',
+  decelerateMid: 'cubic-bezier(0.1,0.9,0.2,1)',
+};
+
+/* end of shared values */
+
 /**
  * Styles for the root slot
  */
 const useStyles = makeStyles({
   // TODO: add themed styles
   root: {
-    ...shorthands.border('1px', 'solid', '#d1d1d1'),
-    borderBottomColor: '#616161',
-    ...shorthands.borderRadius('4px'),
-    display: 'flex',
-    alignItems: 'center',
+    ...shorthands.border('1px', 'solid', 'transparent'),
+    ...shorthands.borderRadius(tokens.borderRadiusMedium),
+    boxSizing: 'border-box',
+    position: 'relative',
 
-    '&:focus-within': {
+    // windows high contrast mode focus indicator
+    ':focus-within': {
       outlineWidth: '2px',
       outlineStyle: 'solid',
       outlineColor: 'transparent',
-      borderBottomColor: '#0078d4',
-      borderBottomWidth: '2px',
     },
-  },
 
-  expandIcon: {
-    flexGrow: 0,
-    flexShrink: 0,
-    ...shorthands.padding('4px'),
+    // bottom focus border, shared with Input, Select, and SpinButton
+    '::after': {
+      boxSizing: 'border-box',
+      content: '""',
+      position: 'absolute',
+      left: '-1px',
+      bottom: '-1px',
+      right: '-1px',
+      height: `max(2px, ${tokens.borderRadiusMedium})`,
+      borderBottomLeftRadius: tokens.borderRadiusMedium,
+      borderBottomRightRadius: tokens.borderRadiusMedium,
+      ...shorthands.borderBottom('2px', 'solid', tokens.colorCompoundBrandStroke),
+      clipPath: 'inset(calc(100% - 2px) 0 0 0)',
+      transform: 'scaleX(0)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.ultraFast,
+      transitionDelay: motionCurves.accelerateMid,
+    },
+    ':focus-within:after': {
+      transform: 'scaleX(1)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.normal,
+      transitionDelay: motionCurves.decelerateMid,
+    },
+    ':focus-within:active:after': {
+      borderBottomColor: tokens.colorCompoundBrandStrokePressed,
+    },
   },
 
   content: {
-    flexGrow: 1,
-    backgroundColor: 'transparent',
-    ...shorthands.border('none'),
+    alignItems: 'center',
```

Comment:
Nit: Consider also setting alignContent.  My understanding is there are cases where alignContent is meaningful when the items don't take up all the available space.

Reply:
`alignContent` affects multi-line flex alignment, which isn't allowed in the combobox trigger/faceplate. I added an explicit `flexWrap: nowrap` to the styles to make that clearer 👍